### ISSUE TITLE
TODO: Listing all variables in a dialog

### ISSUE THREAD

**Issue by thopiekar** [thopiekar/CuraOpenSCADIntegrationPlugin#1]
.. so the user can change them and load her/his model into Cura without changes.

Inspiring can be the ImagePlugin, _BUT_ the dialog over there is hardcoded and I have no idea how to make the dialog creation dynamic. Maybe this needs to be coded in QML.

@fieldOfView: Ping

**Comment by fieldOfView** [comment]
Can you create a list of the variables in Python? If so you can pass that list to QML and use it as a model.

Whatever you do, do not take that image reader plugin as an example! ;-/

**Comment by thopiekar** [comment]
Yes, the patched python module I have in this repository here is able to parse variables out of openscad files, which are designed for thingiverse's Customizer.
So instead of opening a file with it's settings, I would like to show a dialog first, where people can set their settings.

What would you recommend as an better example? Would you recommend to do this in another way?

**Comment by fieldOfView** [comment]
I would say postprocessing plugin, though I don't like how it uses themed controls.
